core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AbstractAnnotationScanner.java

```diff
@@ -104,18 +105,22 @@ private static boolean profileIncluded(OpenApiConfig config, Set<String> profile
     }
 
     public String[] getDefaultConsumes(AnnotationScannerContext context, MethodInfo methodInfo) {
-        if (methodInfo.returnType().kind().equals(Type.Kind.PRIMITIVE)
-                || methodInfo.returnType().name().equals(DotName.createSimple(String.class))) {
+        if (isPrimimive(methodInfo.returnType())) {
```

Comment:
I didn't noticed this last time, but shouldn't it be looking at one of the method's args to determine the "consumes" type?

Reply:
Good point. I'll have to check with the RESTEasy maintainers as to what JAX-RS does, as we are trying to align the defaults there. Can we do that as a separate effort ?

Reply:
Yeah, separate effort is fine with me. I'd suggest leaving out the ability to configure primitive consumes for now though since it definitely won't work based on the return type. Wdyt?

Reply:
As discussed - remove consumes from this text/plain check. consumes default stays json as it was.